multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/OperationsApiServiceImpl.java

```diff
@@ -214,14 +214,29 @@ private Operation getOperation(String operationId) {
             Operation operation = operationService.createQuery()
                                                   .processId(operationId)
                                                   .singleResult();
-            operation = operationsHelper.addState(operation);
+            operation = addOperationState(operation);
             operation = operationsHelper.addErrorType(operation);
             return operation;
         } catch (NoResultException e) {
             throw new NotFoundException(e, Messages.OPERATION_0_NOT_FOUND, operationId);
         }
     }
 
+    private Operation addOperationState(Operation operation) {
+        long startTime = System.nanoTime();
+        operation = operationsHelper.addState(operation);
+        long totalTimeInNano = System.nanoTime() - startTime;
+        LOGGER.debug("Operation state calculation took {} nanoseconds.", totalTimeInNano);
+        if (operation.getCachedState() == null) {
+            LOGGER.warn("Cached operation is null for process \"{}\"!", operation.getProcessId());
+        }
+        if (operation.getCachedState() != operation.getState()) {
+            LOGGER.warn("Operation cached and computed state differ for process \"{}\"! Cached state: \"{}\"; Computed state: \"{}\".",
```

Comment:
Maybe it will be better to use MessageFormatter and extract message in constant. As additon, what will see as status in the cli if states are different?